test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -597,13 +597,12 @@ protected boolean preserveTemplatesUponCompletion() {
      */
     protected boolean resetFeatureStates() {
         try {
-            final Version minimumNodeVersion = minimumNodeVersion();
             if (clusterHasFeature(RestTestLegacyFeatures.FEATURE_STATE_RESET_SUPPORTED) == false) {
                 return false;
             }
 
             // ML reset fails when ML is disabled in versions before 8.7
-            if (isMlEnabled() == false && minimumNodeVersion.before(Version.V_8_7_0)) {
+            if (isMlEnabled() == false && minimumNodeVersion().before(Version.V_8_7_0)) {
```

Comment:
This will be changed after https://github.com/elastic/elasticsearch/pull/102496 is merged